plugin/set.go

```diff
@@ -106,11 +108,39 @@ func (i *Set) Run() error {
 	return i.RunCommand(args...)
 }
 
+func useProtobuf(args ...string) []string {
+	protobufPos := -1
+	for i, arg := range args {
+		if arg == "--protobuf" {
+			protobufPos = i
+			break
+		}
+	}
+
+	if protobufPos == -1 {
+		return args
+	}
+
+	rpc.UseProto = true
+
+	if protobufPos == 0 {
+		return args[1:]
+	}
+
+	if protobufPos == len(args)-1 {
+		return args[:len(args)-1]
+	}
+
+	return append(args[:protobufPos], args[protobufPos+1:]...)
+}
+
 func (i *Set) RunCommand(args ...string) error {
 	if len(args) < 1 {
 		return fmt.Errorf("needs at least one argument")
 	}
 
+	args = useProtobuf(args...)
```

Comment:
Any particular reason why you chose the flag parsing logic above as opposed to adding a flag.Parse(args) when the issued sub command is "start".  

Then inside of the Start function when you construct the server you can specify that it should use protobuf for serialization. You will need to add a new filed to the server to track this. It can be a bool if you go the UseProtoBuf name or a string if you go ProtcolVersion. This way in the common rpc code the server can check if the field is set instead of the global `rpc.UseProto` variable.

Reply:
So `flag` I thought of using it yes, but in the end I went with this approach as the arguments are passed, and `flag.Parse` works on the raw arguments from `os.Args`, besides I think the order had some impact on what was returned, so I thought might as well do something ad-hoc that'd let us place the option wherever.
Besides that's not something that will be used outside Packer, and that'll possibly disappear someday (when Gob is out of the picture), so I thought it was maybe easier to do it this way rather than change how we manage flags in there today.

That being said there are merits to that option of refactoring to handle flags properly, just thought that today was not the day for that, but it should be feasible. Let me know if you think this'd be valuable and I can probably spend some time on that option.

Reply:
Ah I always confuse these args given the whole Run, RunCommand methods our CLI uses. Thanks for the details on why you wrote it this way. I don't think it is important at this very moment given that there is only one flag. But you wrote a decent snippet of code so I was curious why. Sorry for not looking at Run() too much.  

When ready I guess we can add the flag.Parse to the Run method and just create a variable for the UseProto flag.